beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -106,16 +108,37 @@ func (f *ForkChoice) ProcessAttestation(ctx context.Context, validatorIndices []
 }
 
 // InsertOptimisticBlock processes a new block by inserting it to the fork choice store.
-func (f *ForkChoice) InsertOptimisticBlock(
-	ctx context.Context,
-	slot types.Slot,
-	blockRoot, parentRoot, payloadHash [fieldparams.RootLength]byte,
-	justifiedEpoch, finalizedEpoch types.Epoch,
-) error {
+func (f *ForkChoice) InsertOptimisticBlock(ctx context.Context, state state.ReadOnlyBeaconState, root [32]byte) error {
```

Comment:
This function name becomes a bit weird given the main argument is the state. I don't mind it though

Reply:
sure!